### PR TITLE
Move summary function to trait, reduce testbase redundancy, cleanup

### PR DESCRIPTION
**Pull request by jamesiarmes**
Re-roll of #180.

### CODE REVIEW

File: src/Core/ExecutablePluginTrait.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait ExecutablePluginTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getLabelValue() {
```
Comment (milkovsky) [inline src/Core/ExecutablePluginTrait.php]: Where do you inherit from?

File: src/Core/ExecutablePluginTrait.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait ExecutablePluginTrait {
+
+  /**
+   * {@inheritdoc}
```
Comment (milkovsky) [inline src/Core/ExecutablePluginTrait.php:18]: Are you sure here should be {@inheritdoc}? There is no "extend" or "implement" for the trait.

File: src/Core/ExecutablePluginTrait.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      throw new InvalidPluginDefinitionException('The label is not defined for plugin ' . $this->getPluginId() . ' (' . __CLASS__ . ').');
+    }
+    return $definition['label'];
+  }
+
+  /**
+   * Get the translated summary from the label annotation.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\ContextException
+   *   Thrown if a summary was not set.
```
Comment (klausi) [inline src/Core/ExecutablePluginTrait.php:32]: Where is that coming from? You mean it throws a InvalidPluginDefinitionException?

**Comment by klausi** [comment]
merged with further improvements, thanks!

**Pull request closed without merging**